src/main/java/decodes/datasource/ShefPMParser.java

```diff
@@ -191,11 +133,85 @@ else if (dateField.length() == 8)
 		msg.setPM(GoesPMParser.MESSAGE_TIME, new Variable(msgTime));
 		Logger.instance().debug3("SHEF Message Date=" + ccyymmdd.format(msgTime));
 		msg.setPM(PM_TIMEZONE, new Variable(tz.getID()));
-		msg.setPM(GoesPMParser.MESSAGE_LENGTH, new Variable(data.length - headerEnd));
+		msg.setPM(GoesPMParser.MESSAGE_LENGTH, new Variable(data.length));
 		msg.setPM(GoesPMParser.FAILURE_CODE, new Variable('G'));
 		msg.setPM(GoesPMParser.SITE_NAME, new Variable(stationName));
 	}
 
+
+	private Date parseDate(String dateField, TimeZone tz) throws Exception {
```

Comment:
Sorry, I didn't catch this before. the opening `{` needs to be on the next line.